src/test/java/org/junitpioneer/testkit/assertion/PioneerAssert.java

```diff
@@ -88,86 +88,110 @@ public void hasNoReportEntries() {
 
 	@Override
 	public TestCaseStartedAssert hasSingleStartedTest() {
-		return assertSingleTest(actual.testEvents().started().count());
+		Events testEvents = actual.testEvents();
+		assertSingleTest(testEvents.started().count());
+		// Retain all events for returned Assert because TestCaseStartedAssert allows further filtering
+		// (would otherwise erroneously filter twice)
+		return new TestCaseAssertBase(testEvents);
 	}
 
 	@Override
 	public TestCaseFailureAssert hasSingleFailedTest() {
-		return assertSingleTest(actual.testEvents().failed().count());
+		return assertSingleTest(actual.testEvents().failed());
 	}
 
 	@Override
 	public void hasSingleAbortedTest() {
-		assertSingleTest(actual.testEvents().aborted().count());
+		assertSingleTest(actual.testEvents().aborted());
 	}
 
 	@Override
 	public void hasSingleSucceededTest() {
-		assertSingleTest(actual.testEvents().succeeded().count());
+		assertSingleTest(actual.testEvents().succeeded());
 	}
 
 	@Override
 	public void hasSingleSkippedTest() {
-		assertSingleTest(actual.testEvents().skipped().count());
+		assertSingleTest(actual.testEvents().skipped());
 	}
 
 	@Override
 	public TestCaseStartedAssert hasSingleDynamicallyRegisteredTest() {
-		return assertSingleTest(actual.testEvents().dynamicallyRegistered().count());
+		Events testEvents = actual.testEvents();
+		assertSingleTest(testEvents.dynamicallyRegistered().count());
+		// Retain all events for returned Assert because TestCaseStartedAssert allows further filtering
```

Comment:
The double filtering was intentional. Could you provide your rationale for removing it?

Reply:
The comment was actually a bit misleading; I have now adjusted it.

The issue was that previously (in the current version of this project) the `TestCaseAssertBase` was always created with all test / container events, without being restricted to the type of the `hasSingle...()` method:
https://github.com/junit-pioneer/junit-pioneer/blob/f36ee4c351d043e59878fe829303848f5a85e67c/src/test/java/org/junitpioneer/testkit/assertion/PioneerAssert.java#L119-L122

That happened to work because `TestCaseAssertBase.throwable()` previously always restricted the events to `failed()`, however this caused issues for this extension when I implemented `TestCaseAbortedAssert`.

Therefore most `hasSingle...()` now properly filter the events, but for the methods where this is not possible (and which therefore behave the same way they did before) I added those comments.